grodecoder.py

```diff
@@ -923,6 +922,142 @@ def count_remove_ion_solvant(
+            graph = add_attributes_to_nodes(graph, index_resID)
+            list_graph.append(graph)
+
+        atom_id, res_id = [], []
+        for subgraph in list_graph:
+            residue_pairs = zip(
+                nx.get_node_attributes(subgraph, "residue_id").values(),
+                nx.get_node_attributes(subgraph, "residue_name").values(),
+            )
+            residue_pairs_dict = dict(residue_pairs)
+
+            res_id.extend([key for key in sorted(residue_pairs_dict)])
+            atom_id.extend(nx.get_node_attributes(subgraph, "atom_id").values())
+
+        res_id_interval = get_intervals(res_id)
+        atom_id_interval = get_intervals(atom_id)
+
+        counts[list_graph[0]] = {
+            "res_id": res_id,
+            "res_id_interval": res_id_interval,
+            "atom_id": atom_id,
+            "atom_id_interval": atom_id_interval,
+            "name": name,
+            "graph": res_count,
+            "ion": False,
+            "solvant": False,
+            "lipid": True,
+        }
+        # Here we remove all the resIDS (from selected_res_ids) from this universe
+        for interval in res_id_interval:
+            start_end = interval.split("-")
+            if len(start_end) == 1:
+                selection = f"not (resname {alias} and resid {start_end[0]})"
+            else:
+                selection = f"not (resname {alias} and resid {start_end[0]}:{start_end[1]})"
+
+            universe = universe.select_atoms(f"{selection}")
+    return (universe, counts)
+
+
+def count_remove_lipid(
+    universe: mda.core.universe.Universe, input_filepath: str,
+) -> tuple[mda.core.universe.Universe, dict[nx.classes.graph.Graph, dict[str, int]]]:
+    """Count and remove lipid from the MDAnalysis Universe return by
+    the function find_ion_lipid().
+
+    Parameters
+    ----------
+        universe : MDAnalysis.core.universe.Universe
+            MDAnalysis Universe object representing the system.
+        input_filepath : str
+            Path to the input file.
+
+    Returns
+    -------
+        tuple
+            Containing :
+                - the new Universe without lipid.
+                - a dictionary where
+                    - the key is a graph
+                    - and the value is an other dictionary with: atom_start, atom_end, name of the lipid, the counts of removed lipid
+    """
+    counts = {}
+    logger.info("Searching lipid...")
+    lipid_MAD = MAD_DB[MAD_DB["Category"].str.contains("Lipids", case=False, na=False)]
+    
+    for lipid in lipid_MAD.values:
+        universe, counts = find_lipids(lipid.tolist(), universe, counts)
+
+    # Write the new universe without ions and solvant into a new file
+    output_file = f"{Path(input_filepath).stem}_without_H_ions_solvant{Path(input_filepath).suffix}"
+    universe.atoms.write(output_file, reindex=False)
+    logger.debug(
+        f" New structure file without hydrogens, ions and solvants, lipids : {output_file}"
+    )
```

Comment:
Why don't need to save this structure.

Reply:
Because we remove it from the system (after we counted it), and avoid to count it in the get_atom_pair

Reply:
Yes why saving the structure on disk? It's not useful.